### PR TITLE
Redacted URL in logs / errors

### PR DESCRIPTION
**Pull request by dany74q**
So that basic auth creds, if any, won't show up in logs

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,5 +51,5 @@ workflows:
       - run-tests:
           matrix:
             parameters:
-              go-version: ["1.14.2"]
+              go-version: ["1.15.15"]
```
Comment (dany74q) [inline .circleci/config.yml:54]: Yeah... Apparently the url#Redacted() method was introduced in go 1.15; so it makes this a bigger change than I'd anticipated.

I guess I can just copy go's implementation aside if bumping the go version is unacceptable (understandably, of course)

Comment (dany74q) [reply .circleci/config.yml]: Reverted it and just copied the implementation, no need to potentially break existing users

**Review by manicminer — APPROVED** [review]
Many thanks for this fix @dany74q, this LGTM 👍

**Comment by eastebry** [comment]
Hey @dany74q - we've decided to retroactively issue a CVE for this issue. If you would like credit for the discovery or fix, please email security[at]hashicorp.com with the name/identifier you would like us to use (Github user dany74q, Dany from _SomeCompany_, etc.) An acknowledgement will be included in a security bulletin published to https://discuss.hashicorp.com/c/security/

Thank you again!

**Pull request merged**